lib/dao.js

```diff
@@ -148,7 +148,8 @@ function noCallback(err, result) {
  *   - instance (null or Model)
  */
 DataAccessObject.create = function (data, options, cb) {
-  if (stillConnecting(this.getDataSource(), this, arguments)) return;
+  var p;
+  if (p = stillConnecting(this.getDataSource(), this, arguments)) return p;
```

Comment:
Is there any particular reason why you are initializing `p` inside the `if` condition, considering that the variable is declared outside `if`? Linters will complain that your code may contain a typo, because many cases where `if` contains a single `=` were meant to use `==` or `===`.

I strongly prefer the following style:

```js
var p = stillConnecting(this.getDataSource(), this, arguments);
if (p) return p;
```

Also the name `p` is rather non-descriptive, how about `connectedPromise` or `promiseIfConnecting`?